### PR TITLE
Grpc server uses provided host name/ip instead of localhost if possible

### PR DESCRIPTION
**Pull request by fesun**
Now grpc server ignores use provided host name/ip and always return "localhost" as server target. This makes cluster spec propagation propagates "localhost" to remote. See #38519

### CODE REVIEW

**Comment by byronyi** [comment]
Ping @haoyuz; any updates?

**Review by haoyuz — CHANGES_REQUESTED** [review]
Thank you for the PR!

File: tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc [hunk 1]
```diff
@@ -175,7 +180,9 @@ Status GrpcServer::Init(const GrpcServerOptions& opts) {
   // otherwise if 'task_index=-1' the program will abort.
 
   int requested_port;
-  TF_RETURN_IF_ERROR(GetPort(server_def_, &requested_port));
+  string host_name;
```
[the remark this answers is not in the document]
Comment (fesun) [reply tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:183]: To align requested_port, I will host_name_ if you'd prefer.

**Review by haoyuz — APPROVED** [review]

**Comment by haoyuz** [comment]
Adding @bramandia to take another look (assigned in the original issue #38519).

**Comment by gbaned** [comment]
@bramandia Can you please review this PR ? Thanks!

**Comment by fesun** [comment]
@bramandia Could you please take a look at this? Thanks!

**Review by bramandia — APPROVED** [review]

**Comment by bramandia** [comment]
> @bramandia Could you please take a look at this? Thanks!
Done.

Sorry for the delay, this went to my personal email instead.

**Pull request merged**